presto-main/src/main/java/com/facebook/presto/sql/analyzer/AggregationAnalyzer.java

```diff
@@ -313,6 +315,34 @@ protected Boolean visitInPredicate(InPredicate node, Void context)
         protected Boolean visitFunctionCall(FunctionCall node, Void context)
         {
             if (metadata.isAggregationFunction(node.getName())) {
+                if (node.getOrderBy().isPresent()) {
+                    List<Expression> sortKeys = node.getOrderBy().get().getSortItems().stream()
+                            .map(SortItem::getSortKey)
+                            .collect(toImmutableList());
+                    if (node.isDistinct()) {
+                        List<FieldId> argumentReferences = node.getArguments().stream()
+                                .map(NodeRef::of)
+                                .map(columnReferences::get)
+                                .collect(toImmutableList());
+                        for (Expression sortKey : sortKeys) {
+                            if (!argumentReferences.contains(columnReferences.get(NodeRef.of(sortKey)))) {
```

Comment:
Yes, I think it is covered. However, I would rewrite the condition to be more explicit:
```
if (!argumentReferences.contains(columnReferences.get(NodeRef.of(sortKey)) && (!columnReferences.containsNodeRef.of(sortKey) || !fieldIds.contains(columnReferences.get(NodeRef.of(sortKey)))
```
about the fact that `sortKey` might not be in `columReferences`